### ISSUE TITLE
[umbrella] +x flag is not set when using "plain" mode in software section

### ISSUE THREAD

**Issue by Baguage** [cooperative-computing-lab/cctools#1171]
I'm trying to write Umbrella spec for a simple math application, and I ran into a problem in "plain" mode.

Here is my spec:
[bertini.umbrella.txt](https://github.com/cooperative-computing-lab/cctools/files/149039/bertini.umbrella.txt)

When I'm trying to run it, I'm getting

> /bin/sh: /software/bertini: Permission denied

It happens because /software/bertini file does not have +x attribute. The only solution currently is to make file executable in Umbrella cache manually.

I'm sending a pull request with the fix for this problem. I tested it on this example, seems to work.

**Comment by hmeng-19** [comment]
@Baguage , I think the real reason is that the permission of the original source - `/tmp/tmp/BertiniLinux64_v1.5/input` is not set correctly.
Change the permissions of all the mountpoints to be `0755` may introduce some security problems. 
Why not try to change the permission of `/tmp/tmp/BertiniLinux64_v1.5/input` before starting an Umbrella job? Let me know if this works?

**Comment by Baguage** [comment]
Oops, sounds like I missed you response, @hmeng-19 
This happens when I use "plain" mode in software section, so there is no .tar.gz package with permissions. 

I will review my pull request again, but I believe it only sets permissions to 755 for software mountpoints and only in "plain" mode. Without doing that, "plain" mode is effectively broken - executable file will be downloaded, but won't run.

**Comment by hmeng-19** [comment]
@Baguage , in "plain" mode, umbrella copies the file into its own cache and does not really mess up with its access permission. I can not really can reproduce this problem.

I would suggest create a `.tar.gz` file for each software dependency, and put binary into its `bin` subdir, and libraries (if any) into its `lib` subdir.

**Comment by hmeng-19** [comment]
@Baguage , #1579 should fix the problem here. Let me know if it works for you.

**Comment by hmeng-19** [comment]
@Baguage , #1579 has been merged into the master. Please test the master to see whether it works for you.
